src/sidebar/services/groups.js

```diff
@@ -137,9 +137,40 @@ function groups($rootScope, store, api, isSidebar, localStorage, serviceUrl, ses
         params.document_uri = uri;
       }
       documentUri = uri;
-
-      // Fetch groups from the API.
-      return api.groups.list(params, null, { includeMetadata: true });
+      
+      if (features.flagEnabled('community_groups')) {
+        return $q.all([api.profile.groups.read({expand: "organization"}),
+               api.groups.list(params, null, { includeMetadata: true })]
+        ).then(data => {
+          const myGroups = data[0];
```

Comment:
I agree. As far as I can tell the data doesn't actually contain any more info, it's really just the token that get's returned that is of interest but I feel like there's probably a better way to do that.

Reply:
> As far as I can tell the data doesn't actually contain any more info, it's really just the token that get's returned that is of interest but I feel like there's probably a better way to do that.

The short answer is that I agree with both of you 🙂.

Having the structure of the result returned from API calls change depending on parameters is not something I like. The reason it was done that way was because the original API of the `api` service did not support returning any of the metadata from the response (headers, status code etc. - anything other than the JSON body) and I wanted to be able to get this metadata here without needing to change all other uses of the `api` service in the same commit. Having calls to methods on the `api` service return just the JSON body is convenient in most cases since none of our API usage requires looking at response headers or differentiating between status codes beyond "success" (2xx) and "failure" (4xx, 5xx). However I think we will probably need to change that at some point so that all calls to API methods return an object with `data`, `status`, `headers` properties etc.

Why was the token needed at all? The business requirement is that the "Public" group be excluded from the groups list if the user is logged out and included if they are logged in. One way to do this is to check whether the request to the `/api/groups` endpoint was authenticated or not. In theory this is nice because the users log-in state could change while the request is being made and this tells us whether or not that specific request was authenticated. However whenever the user logs in we'll trigger another refresh of the groups list anyway.

An alternative would be to check whether the user is logged in when the response is processed (`store.profile().userid !== null`). That would ensure that the filtering behaviour matched the UI state.